cli/src/main.rs

```diff
@@ -440,19 +442,30 @@ async fn main() -> Result<()> {
             cancel,
         } => {
             let drain = !cancel;
-            for drone in drone {
+            for ref drone in drone {
                 nats.request(&DrainDrone {
                     cluster: ClusterName::new(&cluster),
-                    drone: DroneId::new(drone),
+                    drone: DroneId::new(drone.clone()),
                     drain,
                 })
-                .await?;
-            }
-
-            if drain {
-                println!("{}", "Draining started on drone.".bright_green());
-            } else {
-                println!("{}", "Draining cancelled on drone.".bright_green());
+                .await
+                .map_or_else(
```

Comment:
the `map` here makes it harder to follow. I would avoid `map*` when the result is not actually needed.